### PR TITLE
Added Github action for YugabyteDB

### LINKED ISSUE #5098: Github action to bring up a single node yugabyte cluster for automated tests
Similar to https://github.com/marketplace/actions/setup-postgresql-with-postgresql-extensions-and-unprivileged-user and other github marketplace actions for other databases, it would be nice to have a standard github action that brings up yugabyte for clients to test against. This could use the docker command line documented at https://docs.yugabyte.com/latest/quick-start/create-local-cluster/docker/

### PR DESCRIPTION
**Pull request by baba230896**
### Summary:

- Added Github action for YugabyteDB

Fixes: https://github.com/yugabyte/yugabyte-db/issues/5098

### CODE REVIEW

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    - name: Print output variables
+      run: |
+        echo "Container ID: ${{ steps.server.outputs.container_id }}"
+        echo "YB Master UI Port: ${{ steps.server.outputs.yb_master_ui_port }}"
+        echo "YB Tserver UI Port: ${{ steps.server.outputs.yb_tserver_ui_port }}"
+        echo "YSQL Port: ${{ steps.server.outputs.ysql_port }}"
+        echo "YCQL Port: ${{ steps.server.outputs.ycql_port }}"
+
+    - name: Test YSQL API
+      run: docker run --network host --rm yugabytedb/yugabyte-client:latest ysqlsh -h localhost -p "${{ steps.server.outputs.ysql_port }}" -c \\conninfo
```
Comment (iSignal) [inline .github/workflows/main.yaml]: we can do a `CREATE TABLE foo(id int primary key); INSERT INTO foo SELECT * FROM generate_series(1,10);` in the -c

File: .github/workflows/main.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    - name: Test YCQL API
+      run: docker run --network host --rm yugabytedb/yugabyte-client:latest ycqlsh localhost "${{ steps.server.outputs.ycql_port }}" --execute 'SHOW HOST'
```
Comment (iSignal) [inline .github/workflows/main.yaml]: `create keyspace foo; create table foo.bar(id int primary key); insert into foo.bar (id) values (1);`

File: entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+SERVICE_ID=$(docker run \
+  -p "${INPUT_YB_MASTER_UI_PORT}:${INPUT_YB_MASTER_UI_PORT}" \
+  -p "${INPUT_YB_TSERVER_UI_PORT}:${INPUT_YB_TSERVER_UI_PORT}" \
+  -p "${INPUT_YSQL_PORT}:${INPUT_YSQL_PORT}" \
+  -p "${INPUT_YCQL_PORT}:${INPUT_YCQL_PORT}" \
+  --detach \
+  yugabytedb/yugabyte:"${INPUT_YB_IMAGE_TAG}" \
+  bin/yugabyted start --ycql_port="${INPUT_YCQL_PORT}" --ysql_port="${INPUT_YSQL_PORT}" \
+  --master_webserver_port="${INPUT_YB_MASTER_UI_PORT}" --tserver_webserver_port="${INPUT_YB_TSERVER_UI_PORT}" \
+  --daemon=false)
```
Comment (iSignal) [inline entrypoint.sh]: let's also add a `--ui=false` (even though it is the default)

File: action.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    description: 'YSQL API Port'
+    required: false
+    default: '5433'
+  ycql_port:
+    description: 'YCQL API Port'
+    required: false
+    default: '9042'
+outputs:
+  container_id: # id of output
+    description: 'Contianer ID'
```
Comment (iSignal) [inline action.yml]: Container

File: .github/workflows/main.yaml [hunk 5]
```diff
@@ -0,0 +1,34 @@
+name: Test yugabyte-db-action
+
+on:
+  push:
+    branches:
+      - master
```
Comment (iSignal) [inline .github/workflows/main.yaml]: We should fire the repo dispatch action from the main repo and test each time a new version comes out.

File: .github/workflows/main.yaml [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+jobs:
+  test-yugabytedb-action:
+    runs-on: ubuntu-latest
```
Comment (iSignal) [inline .github/workflows/main.yaml:12]: would be good to also run it on centos:latest

File: Dockerfile [hunk 7]
```diff
@@ -0,0 +1,7 @@
+FROM docker:stable
+
+RUN set -eux ; \
+    apk add  --no-cache bash
+COPY entrypoint.sh /usr/bin/entrypoint
```
Comment (iSignal) [inline Dockerfile:5]: I guess there's no way to define new github actions with multiple steps without using a Docker container?

File: .github/workflows/main.yaml (hunk 5 above)
Comment (baba230896) [reply .github/workflows/main.yaml]: Okay. I'll update the required changes and create the PR for the `yugabyte-db` repo as well for repository dispatch events.

File: .github/workflows/main.yaml (hunk 6 above)
Comment (baba230896) [reply .github/workflows/main.yaml:12]: Github only supports macOS, windows, and ubuntu as an OS for Github actions. 
Reference Doc: https://docs.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners

**Comment by baba230896** [comment]
@iSignal 
I've raised the PR for the repository dispatch event. Please take a look at it.
https://github.com/yugabyte/yugabyte-db/pull/5349

**Review by iSignal — APPROVED** [review]

**Pull request merged**